hudi-cli/src/test/java/org/apache/hudi/cli/testutils/HoodieTestCommitMetadataGenerator.java

```diff
@@ -90,8 +91,7 @@ public static void createCommitFileWithMetadata(String basePath, String commitTi
     for (String name : commitFileNames) {
       HoodieCommitMetadata commitMetadata =
               generateCommitMetadata(basePath, commitTime, fileId1, fileId2, writes, updates, extraMetadata, true);
-      String content = commitMetadata.toJsonString();
-      createFileWithMetadata(basePath, configuration, name, content);
+      createFileWithMetadata(basePath, configuration, name, serializeCommitMetadata(convertCommitMetadata(commitMetadata)).get());
     }
```

Comment:
The combination of `serializeCommitMetadata(convertCommitMetadata` looks everywhere, can we put the conversion of `convertCommitMetadata` inside `serializeCommitMetadata`